src/Services/FHIR/FhirServiceBase.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace OpenEMR\Services\FHIR;
+
+use OpenEMR\Validators\ProcessingResult;
+
+/**
+ * Base class for FHIR Service implementations.
+ *
+ * Implementations are required to override the following methods:
+ * - loadSearchParameters
+ * - parseOpenEMRRecord
+ * - parseFhirResource
+ * - insertOpenEMRRecord
+ * - updateOpenEMRRecord
+ * - getOne
+ * - searchForOpenEMRRecords
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Dixon Whitmire <dixonwh@gmail.com>
+ * @copyright Copyright (c) 2020 Dixon Whitmire <dixonwh@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+abstract class FhirServiceBase
```

Comment:
I added a base class for the FHIR services. Hopefully the docs are self explanatory.